[mono] Move some culture-dependent sources to a separate directory

monojenkins opened the pull request:
!! This PR is a copy of mono/mono#19912,  please do not edit or review it in this repo !!<br/>Do not automatically approve this PR:<br/><br/>* Consider how the changes affect configurations in this repo,<br/>* Check effects on files that are not mirrored,<br/>* Identify test cases that may be needed in this repo.<br/><br/>!! Merge the PR only after the original PR is merged !!<br/><hr/><br/>We'd like to remove culture-info-tables.h from the dotnet/runtime copy of mono since we use ICU
there for globalization like the rest of .NET 5.

The plan is:

1. We move the file (and some others) to a separate directory (this PR)
2. We tell the sync bot to skip culture/ in the future
3. We delete the files (or make them empty) in dotnet/runtime

lambdageek commented:
Hmm. The bot synced culture/ over.  Maybe we let it do its thing, then disable it, and in a follow-up PR remove the files.  It should be enough just for the empty directory to exist, I think.

lambdageek committed:
[mono] Move some culture-dependent source files to a separate directory

We'd like to remove culture-info-tables.h from the dotnet/runtime copy of mono since we use ICU
there for globalization like the rest of .NET 5.

The plan is:
1. We move the file (and some others) to a separate directory (this PR)
2. We tell the sync bot to skip `culture/` in the future
3. We delete the files (or make them empty) in dotnet/runtime
